### PR TITLE
usm: Rename assets to usm

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?
Changes the compilation artifacts of USM to be called `usm.o` instead of `http.o`

### Motivation
We're trying to better reflect USM, as now it is more than just http capturing.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by usamasaqib — APPROVED** [review]

**Review by nplanel — APPROVED** [review]

**Review by akarpz — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -326,7 +326,7 @@
 /pkg/network/usm/                       @DataDog/universal-service-monitoring
 /pkg/ebpf/                              @DataDog/ebpf-platform
 /pkg/ebpf/bytecode/runtime/conntrack.go @DataDog/Networks @DataDog/universal-service-monitoring
-/pkg/ebpf/bytecode/runtime/http.go      @DataDog/Networks @DataDog/universal-service-monitoring
+/pkg/ebpf/bytecode/runtime/usm.go      @DataDog/Networks @DataDog/universal-service-monitoring
```
Comment (DanielLavie) [inline .github/CODEOWNERS:329]: Add missing space

File: .gitignore [hunk 2]
```diff
@@ -115,6 +115,7 @@ compile_commands.json
 pkg/ebpf/bytecode/build/**/*.d
 pkg/ebpf/bytecode/runtime/conntrack.go
 pkg/ebpf/bytecode/runtime/http.go
```
Comment (DanielLavie) [inline .gitignore:117]: Do we still need this?

File: pkg/network/usm/compile.go [hunk 3]
```diff
@@ -14,11 +14,11 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/process/statsd"
 )
 
-//go:generate go run ../../../pkg/ebpf/include_headers.go ../../../pkg/network/ebpf/c/runtime/http.c ../../../pkg/ebpf/bytecode/build/runtime/http.c ../../../pkg/ebpf/c ../../../pkg/ebpf/c/protocols ../../../pkg/network/ebpf/c/runtime ../../../pkg/network/ebpf/c
-//go:generate go run ../../../pkg/ebpf/bytecode/runtime/integrity.go ../../../pkg/ebpf/bytecode/build/runtime/http.c ../../../pkg/ebpf/bytecode/runtime/http.go runtime
+//go:generate go run ../../../pkg/ebpf/include_headers.go ../../../pkg/network/ebpf/c/runtime/usm.c ../../../pkg/ebpf/bytecode/build/runtime/usm.c ../../../pkg/ebpf/c ../../../pkg/ebpf/c/protocols ../../../pkg/network/ebpf/c/runtime ../../../pkg/network/ebpf/c
+//go:generate go run ../../../pkg/ebpf/bytecode/runtime/integrity.go ../../../pkg/ebpf/bytecode/build/runtime/usm.c ../../../pkg/ebpf/bytecode/runtime/usm.go runtime
 
 func getRuntimeCompiledHTTP(config *config.Config) (runtime.CompiledOutput, error) {
```
Comment (DanielLavie) [inline pkg/network/usm/compile.go:20]:
```suggestion
func getRuntimeCompiledUSM(config *config.Config) (runtime.CompiledOutput, error) {
```

**Review by DanielLavie — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (guyarb) [reply .gitignore:117]: yes. after merging this change, the local instance might be committed accidentally to the main branch

File: pkg/network/usm/compile_test.go [hunk 4]
```diff
@@ -18,13 +18,13 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/kernel"
 )
 
-func TestHttpCompile(t *testing.T) {
+func TestUSMCompile(t *testing.T) {
 	if !rtcHTTPSupported(t) {
```
Comment (DanielLavie) [inline pkg/network/usm/compile_test.go:22]:
```suggestion
	if !rtcUSMSupported(t) {
```

**Review by DanielLavie — CHANGES_REQUESTED** [review]

File: pkg/network/usm/compile_test.go (hunk 4 above)
Comment (guyarb) [reply pkg/network/usm/compile_test.go:22]: no, that's inaccurate as it check the kernel meets the HTTP monitoring requirements

**Review by DanielLavie — APPROVED** [review]

**Review by KSerrania — APPROVED** [review]

**Pull request merged**